platform/ext/target/nordic_nrf/common/core/hw_init.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <nrf.h>
+#include "array.h"
+
+/* This routine clears all ARM MPU region configuration. */
+static void hw_init_mpu(void)
+{
+	int i;
+
+	int num_regions =
+		((MPU->TYPE & MPU_TYPE_DREGION_Msk) >> MPU_TYPE_DREGION_Pos);
+
+	for (i = 0; i < num_regions; i++) {
+		ARM_MPU_ClrRegion(i);
+	}
+}
+
+/* This routine resets Cortex-M system control block components and core
+ * registers.
+ */
+void hw_init_reset_on_boot(void)
+{
+	/* Disable interrupts */
+	__disable_irq();
+
+	/* Reset exception and interrupt mask state (PRIMASK handled by
+	 * __enable_irq below)
+	 */
+	__set_FAULTMASK(0);
+	__set_BASEPRI(0);
```

Comment:
@carlescufi Zephyr has similar initialization handling, however it does not handle the BASEPRI.
This likely does not result in any issues for zephyr at the moment because of zephyr actually writing this register during its irq_lock/unlock functions.

However that could be a potential problem in the future, so fixing this in zephyr would be wise IMO.

Reply:
Are you sure that Zephyr doesn't initialize `BASEPRI` at boot? Can you link me to the place where it should be initialized?

Reply:
Turns out it is initialised here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/arm/core/aarch32/cortex_m/reset.S#L114

But not part of the CONFIG_INIT_ARCH_HW_AT_BOOT option.